packages/components/addon/components/hds/tabs/index.js

```diff
@@ -56,6 +56,11 @@ export default class HdsTabsIndexComponent extends Component {
       block: 'nearest',
       inline: 'nearest',
     });
+
+    if (typeof this.args.onTabClick === 'function') {
```

Comment:
for consistency (eg. see the `onClickToggle` in the [Disclosure component](https://design-system-components-hashicorp.vercel.app/utilities/disclosure#component-api)) I would call this callback/hook function `onClickTab`.